pkg/appregistry/appregistry.go

```diff
@@ -0,0 +1,48 @@
+package appregistry
```

Comment:
This whole file could be named as factory.go

Reply:
Yeah, good point. I thought about it. I also like the idea of a `{package name}.go` file which has the entry level function for the given package. `NewFactory` is the entry level function one must call to use the `appregistry` package.